third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompting-checker-unittests.js

```diff
@@ -0,0 +1,288 @@
+        return goog.Promise.resolve(FAKE_APP_1_KNOWN_APP);
+      return goog.Promise.reject();
+    }
+  };
+  /** @type {?} */
+  var mockedConstructor = mockControl.createConstructorMock(
+      PermissionsChecking, 'KnownAppsRegistry');
+  mockedConstructor().$returns(mockInstance);
+}
+
+/**
+ * Sets up mocks for get() and set() methods of the chrome.storage.local
+ * Extensions API.
+ * @param {!goog.testing.MockControl} mockControl
+ * @param {!goog.testing.PropertyReplacer} propertyReplacer
+ * @param {!Object} fakeInitialData Fake data to be returned from get().
+ * @param {Object} expectedDataToBeWritten If non-null, then the expected data
+ * to be passed into set(); if null, then set() isn't expected to be called.
+ */
+function setUpChromeStorageMock(
+    mockControl, propertyReplacer, fakeInitialData, expectedDataToBeWritten) {
+  propertyReplacer.set(
+      chrome, 'storage',
+      {local: {
+         get: mockControl.createFunctionMock('chrome.storage.local.get'),
+         set: mockControl.createFunctionMock('chrome.storage.local.set')}});
+
+  /** @type {?} */ chrome.storage.local.get;
+  chrome.storage.local.get(STORAGE_KEY, ignoreArgument).$does(
+      function(key, callback) {
+        callback(fakeInitialData);
+      });
+
+  /** @type {?} */ chrome.storage.local.set;
+  if (!goog.isNull(expectedDataToBeWritten))
+    chrome.storage.local.set(expectedDataToBeWritten);
+}
+
+/**
+ * Sets up mocks for the GSC.PopupWindow.Server.runModalDialog method.
+ * @param {!goog.testing.MockControl} mockControl
+ * @param {!MockedDialogBehavior} mockedBehavior If |NOT_RUN|, then the method
+ * is not expected to be called; otherwise, the called method will return the
+ * corresponding result.
+ */
+function setUpDialogMock(mockControl, mockedBehavior) {
+  /** @type {?} */
+  var mockedFunction = mockControl.createMethodMock(
+      GSC.PopupWindow.Server, 'runModalDialog');
+  var mockAction;
+  switch (mockedBehavior) {
+    case MockedDialogBehavior.NOT_RUN:
+      return;
+    case MockedDialogBehavior.USER_APPROVES:
+      mockAction = function() {
+        return goog.Promise.resolve(true);
+      };
+      break;
+    case MockedDialogBehavior.USER_DENIES:
+      mockAction = function() {
+        return goog.Promise.resolve(false);
+      };
+      break;
+    case MockedDialogBehavior.USER_CANCELS:
+      mockAction = function() {
+        return goog.Promise.reject();
+      };
+      break;
+  }
+  mockedFunction(ignoreArgument, ignoreArgument, ignoreArgument).$does(
+      mockAction);
+}
+
+/**
+ * Returns a negation of the given promise.
+ * @param {!goog.Promise} promise
+ * @return {!goog.Promise.<undefined>} A promise which is rejected when
+ * |promise| gets resolved, and resolved (with no data) when |promise| is
+ * rejected.
+ */
+function negatePromise(promise) {
+  return promise.then(function() {
+    fail('Unexpected successful response');
+  }, function() {
+    // Do not pass on the rejection.
```

Comment:
I'd have expected a resolve() call here -- does this not just ignore the rejection?

Reply:
Maybe I misunderstood the question, but there's no resolve() method on the promise; and returning undefined from this function should resolve the resulting promise - according to the docs:

"If the Promise is rejected, the onRejected callback will be invoked with the rejection reason as argument, and the child Promise will be resolved with the return value or rejected with the thrown value of the callback."